addons/website_event_track/static/src/js/service_worker.js

```diff
@@ -308,7 +313,7 @@ const prefetchUrls = async (urls = []) => {
             continue;
         }
         try {
-            await processFetchRequest({ request: new Request(url) });
+            await processFetchRequest(new Request(url), true);
```

Comment:
I'm sorry but I'm still not sure about this @pparidans @pyduf

So this is the call we want to track, since it has the "true" argument.
If I grep in the code, that method is called when the action "prefetch-pages" is called.
But at that point, the user has already "rendered" the event page, and we have tracked that call, correct?

What you mean is, once the PWA is installed, we would not track anything anymore if we always add the header here.
That seems ok to me, as the PWA usage is very niche anyway.

Please tell me if I misunderstood anything.

Reply:
Don't forget that, once installed (cf. after the very first page request), a service worker actually works with a "scope" (ie. `/event`) and not a a single page. So every request made under that scope will pass through the service worker's fetch event listener.

In other words, disabling tracking globally in the service worker will then disable tracking for every request made to a subpath of `/event` (aka. the scope)!

Reply:
Actually "installing the app" has nothing to do with the service worker behavior: he'll be present and catching requests anyway.

Having a service worker and a (quite minimal) handling of offline are "only" some of the prerequisite for a website to be seen as installable by the browser.